app/modules/complex_date_time/models.py

```diff
@@ -191,6 +191,83 @@ def isoformat_utc(self):
     def isoformat_in_timezone(self):
         return self.get_datetime_in_timezone().isoformat()
 
+    # used from parameters.py for object which have time/timeSpecifity patches (currently encounter and sighting)
+    @classmethod
+    def patch_replace_helper(cls, obj, field, value):
+        from app.modules.complex_date_time.models import ComplexDateTime, Specificities
+        from datetime import datetime
+        from app.utils import normalized_timezone_string
+        from .models import db
+        import pytz
+
+        # * note: field==time requires `value` is iso8601 **with timezone**
+        # this gets a little funky in the event there is *no existing time set* as the patch
+        #   happens in two parts that know nothing about each other.  so we have to create a ComplexDateTime and
+        #   *fake* the other field value (time/timeSpecificity) upon doing so.  :(  we then hope that the subsequent
+        #   patch for the other field is coming down the pipe.  api user beware!
+        # note: the dict-based all-at-once solution below is the better choice if you can swing it.
+
+        if (field == 'time' or field == 'timeSpecificity') and isinstance(value, str):
+            dt = None
+            specificity = None
+            timezone = None
+            if field == 'time':
+                # this will throw ValueError if not parseable
+                dt = datetime.fromisoformat(value)
+                if not dt.tzinfo:
+                    raise ValueError(f'passed value {value} does not have time zone data')
```

Comment:
Are these Value errors specifically caught in the resources file? I have been taking the philosophy that exceptions should not escape outside of Houston and should be caught and converted to an abort in the resources files.
Not sure if that is a common philosophy for web dev but it does seem cleaner to me.

Reply:
thanks for bringing this up howard; it made me catch something that got swallowed in a rebase or something.  fixed up the patching via a [new commit](https://github.com/WildMeOrg/houston/pull/420/commits/d53894445563b63394c62a3d6f37203ab1056f54).  but yeah basically both Encounter and Sighting `parameters.py` use this for patching.  so the raised errors end up there as a 409 return, they are not caught.  i think this is a better behavior than `abort()` for the sake of patching.